### ISSUE TITLE
 G_loss_frecon = tf.zeros(1)???

### ISSUE THREAD

**Issue by huyanfei-cqupt** [tranluan/Nonlinear_Face_3DMM#48]
@tranluan  
Is G_loss_frecon missing?  I can't find the relevant code in this model_nonlinear_3DMM.py
can you share the code to me? 
Thank you very much!

**Comment by HOMGH** [comment]
> @tranluan
> Is G_loss_frecon missing? I can't find the relevant code in this model_nonlinear_3DMM.py
> can you share the code to me?
> Thank you very much!

Hi, You're right. It seems that perceptual loss (loss_frecon) is missing in the code. Did you find it anywhere?

**Comment by huyanfei-cqupt** [comment]
> 
> 
> > @tranluan
> > Is G_loss_frecon missing? I can't find the relevant code in this model_nonlinear_3DMM.py
> > can you share the code to me?
> > Thank you very much!
> 
> Hi, You're right. It seems that perceptual loss (loss_frecon) is missing in the code. Did you find it anywhere?

I just found this problem. For the model_non_linear_3DMM.py file, I didn't find the corresponding place.

**Comment by tranluan** [comment]
It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)

You also need code and model from:
https://github.com/ZZUTK/Tensorflow-VGG-face

**Comment by huyanfei-cqupt** [comment]
> 
> 
> It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)
> 
> You also need code and model from:
> https://github.com/ZZUTK/Tensorflow-VGG-face

Thank you! @tranluan i will  to try it.

**Comment by HOMGH** [comment]
> It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)
> 
> You also need code and model from:
> https://github.com/ZZUTK/Tensorflow-VGG-face

Thanks @tranluan .

**Comment by huyanfei-cqupt** [comment]
> > It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)
> > You also need code and model from:
> > https://github.com/ZZUTK/Tensorflow-VGG-face
> 
> Thanks @tranluan .

@HOMGH hi, Have you written the part of the code about the perceptual loss frecon? Can you share the code with me?  Thank you very much!

**Comment by HOMGH** [comment]
> > > It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)
> > > You also need code and model from:
> > > https://github.com/ZZUTK/Tensorflow-VGG-face
> > 
> > 
> > Thanks @tranluan .
> 
> @HOMGH hi, Have you written the part of the code about the perceptual loss frecon? Can you share the code with me? Thank you very much!

Hi,Sorry not yet.

**Comment by huyanfei-cqupt** [comment]
> @tranluan
> Is G_loss_frecon missing? I can't find the relevant code in this model_nonlinear_3DMM.py
> can you share the code to me?
> Thank you very much!



> > > > It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)
> > > > You also need code and model from:
> > > > https://github.com/ZZUTK/Tensorflow-VGG-face
> > > 
> > > 
> > > Thanks @tranluan .
> > 
> > 
> > @HOMGH hi, Have you written the part of the code about the perceptual loss frecon? Can you share the code with me? Thank you very much!
> 
> Hi,Sorry not yet.

Well, I tried to imitate the writing of proxy.py and modified it, but some errors appeared. I haven't solved it until now. If you want to see the part of the code I wrote, I can email you, and then we Let's discuss it together.

**Comment by huyanfei-cqupt** [comment]
> > It should be similar losses defined in [model_non_linear_3DMM_proxy.py](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)
> > You also need code and model from:
> > https://github.com/ZZUTK/Tensorflow-VGG-face
> 
> Thanks @tranluan .

Hi,@HOMGH  Have you solved the perceptual loss function code in this model_nonlinear_3DMM.py？

**Comment by LY-Donkey-Burger** [comment]
> > > 它应该是[model_non_linear_3DMM_proxy.py中定义的类似损失您还需要来自： ](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)[https](https://github.com/ZZUTK/Tensorflow-VGG-face) ://github.com/ZZUTK/Tensorflow-VGG-face
> > > 的代码和模型
> > > [](https://github.com/ZZUTK/Tensorflow-VGG-face)
> > 
> > 
> > 谢谢@tranluan.
> 
> 你好，@HOMGH 这个model_nonlinear_3DMM.py中的感知损失函数代码你解决了吗？



> > > 它应该是[model_non_linear_3DMM_proxy.py中定义的类似损失您还需要来自： ](https://github.com/tranluan/Nonlinear_Face_3DMM/blob/03b526cf6c6d72420eccd18b37619d2f5365d87f/model_non_linear_3DMM_proxy.py#L454)[https](https://github.com/ZZUTK/Tensorflow-VGG-face) ://github.com/ZZUTK/Tensorflow-VGG-face
> > > 的代码和模型
> > > [](https://github.com/ZZUTK/Tensorflow-VGG-face)
> > 
> > 
> > 谢谢@tranluan.
> 
> 你好，@HOMGH 这个model_nonlinear_3DMM.py中的感知损失函数代码你解决了吗？

**Comment by LY-Donkey-Burger** [comment]
Hi,did you solve the loss function code?
